google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/EnhancedBigtableStub.java

```diff
@@ -204,10 +205,26 @@ public static ClientContext createClientContext(EnhancedBigtableStubSettings set
             ? ((InstantiatingGrpcChannelProvider) builder.getTransportChannelProvider()).toBuilder()
             : null;
 
-    if (builder.getEnableRoutingCookie() && transportProvider != null) {
-      // TODO: this also need to be added to BigtableClientFactory
-      // patch cookies interceptor
-      transportProvider.setInterceptorProvider(() -> ImmutableList.of(new CookiesInterceptor()));
+    ErrorCountPerConnectionMetricTracker errorCountPerConnectionMetricTracker =
+        new ErrorCountPerConnectionMetricTracker(
+            builder.getBackgroundExecutorProvider().getExecutor(),
```

Comment:
I think we want to get the executor from the client context. We can have your ` startConnectionErrorCountTracker` takes in a executor and start the task after the client context is built. 

So on line 253, we can have:

ClientContext clientContext = ClientContext.create(builder.build());
errorCountPerConnectionMetricTracker.startConnectionErrorCountTracker(clientContext.getExecutor());